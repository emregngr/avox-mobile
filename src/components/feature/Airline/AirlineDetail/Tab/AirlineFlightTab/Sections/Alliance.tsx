import { MaterialCommunityIcons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'

interface AllianceProps {
  alliance: string
  iconColor: string
  title: string
}

export const Alliance = ({ alliance, iconColor, title }: AllianceProps) => (
  <AirlineSectionRow title={title}>
    <View className="items-center my-5">
      <MaterialCommunityIcons color={iconColor} name="shield" size={40} />

      <ThemedText className="mt-2.5" color="text-90" type="h4">
        {alliance}
      </ThemedText>
    </View>
  </AirlineSectionRow>
)
