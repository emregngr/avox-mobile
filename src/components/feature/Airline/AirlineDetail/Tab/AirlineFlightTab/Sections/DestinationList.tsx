import React from 'react'
import { View } from 'react-native'

import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { DestinationRowCard } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Cards/DestinationRowCard'

interface DestinationListProps {
  destinations: string[]
  iconColor: string
  title: string
}

export const DestinationList = ({ destinations, iconColor, title }: DestinationListProps) => (
  <AirlineSectionRow title={title}>
    <View className="flex-row flex-wrap -mx-1">
      {destinations?.map((destination, index) => (
        <DestinationRowCard destination={destination} iconColor={iconColor} key={index} />
      ))}
    </View>
  </AirlineSectionRow>
)
