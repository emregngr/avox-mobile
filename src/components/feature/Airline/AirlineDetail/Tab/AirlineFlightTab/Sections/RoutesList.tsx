import React from 'react'

import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { RouteRowCard } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Cards/RouteRowCard'
import type { AirlineRoute } from '@/types/feature/airline'

interface RoutesListProps {
  iconColor: string
  routes: AirlineRoute[]
  title: string
}

export const RoutesList = ({ iconColor, routes, title }: RoutesListProps) => (
  <AirlineSectionRow title={title}>
    {routes?.map((route, index) => (
      <RouteRowCard
        destinationIata={route.destinationIata}
        iconColor={iconColor}
        key={index}
        origin={route.origin}
      />
    ))}
  </AirlineSectionRow>
)
