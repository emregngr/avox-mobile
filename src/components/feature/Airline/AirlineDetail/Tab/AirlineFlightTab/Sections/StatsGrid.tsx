import React from 'react'
import { View } from 'react-native'

import { StatsCard } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Cards/StatsCard'

interface StatsGridProps {
  iconColor: string
  localeStrings: {
    country: string
    domesticDestinations: string
    internationalDestination: string
    totalDestination: string
  }
  stats: {
    destinationCount: number
    destinationCountries: number
    domesticConnections: number
    internationalConnections: number
  }
}

export const StatsGrid = ({ iconColor, localeStrings, stats }: StatsGridProps) => (
  <>
    <View className="flex-row gap-x-4">
      <StatsCard
        iconColor={iconColor}
        iconName="flag"
        label={localeStrings.totalDestination}
        value={stats.destinationCount}
      />
      <StatsCard
        iconColor={iconColor}
        iconName="earth"
        label={localeStrings.country}
        value={stats.destinationCountries}
      />
    </View>

    <View className="flex-row gap-x-4">
      <StatsCard
        iconColor={iconColor}
        iconName="home"
        label={localeStrings.domesticDestinations}
        value={stats.domesticConnections}
      />
      <StatsCard
        iconColor={iconColor}
        iconName="globe"
        label={localeStrings.internationalDestination}
        value={stats.internationalConnections}
      />
    </View>
  </>
)
