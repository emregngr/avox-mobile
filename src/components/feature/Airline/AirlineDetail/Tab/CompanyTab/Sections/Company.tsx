import React, { useMemo } from 'react'

import { AirlineRowItem } from '@/components/feature/Airline/AirlineDetail/AirlineRowItem'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airline } from '@/types/feature/airline'
import { formatNumber } from '@/utils/feature/formatNumber'

interface CompanyProps {
  airlineData: Airline
}

export const Company = ({ airlineData }: CompanyProps) => {
  const { selectedLocale } = useLocaleStore()

  const { companyInfo, isoCountry, isoRegion, operations } = airlineData || {}
  const { employeeCount, foundingYear, parentCompany, passengerCapacity } = companyInfo || {}
  const { businessModel, businessType, skytraxRating, slogan } = operations || {}

  const localeStrings = useMemo(
    () => ({
      airlineInfo: getLocale('airlineInfo'),
      businessModel: getLocale('businessModel'),
      businessType: getLocale('businessType'),
      isoCountryCode: getLocale('isoCountryCode'),
      isoRegionCode: getLocale('isoRegionCode'),
      millionPerYear: getLocale('million/year'),
      numberOfEmployees: getLocale('numberOfEmployees'),
      numberOfPassengers: getLocale('numberOfPassengers'),
      parentCompany: getLocale('parentCompany'),
      skytraxRating: getLocale('skytraxRating'),
      slogan: getLocale('slogan'),
      year: getLocale('year'),
      yearOfEstablishment: getLocale('yearOfEstablishment'),
    }),
    [selectedLocale],
  )

  return (
    <AirlineSectionRow title={localeStrings.airlineInfo}>
      <AirlineRowItem
        icon="calendar"
        label={localeStrings.yearOfEstablishment}
        value={`${foundingYear} (${new Date().getFullYear() - parseInt(foundingYear)} ${localeStrings.year})`}
      />

      <AirlineRowItem icon="business" label={localeStrings.parentCompany} value={parentCompany} />

      <AirlineRowItem
        icon="people"
        label={localeStrings.numberOfPassengers}
        value={`${passengerCapacity} ${localeStrings.millionPerYear}`}
      />

      <AirlineRowItem
        icon="people"
        label={localeStrings.numberOfEmployees}
        value={formatNumber(employeeCount)}
      />

      <AirlineRowItem
        className="capitalize"
        icon="briefcase"
        label={localeStrings.businessModel}
        value={businessModel?.replace('_', ' ')}
      />

      <AirlineRowItem
        className="capitalize"
        icon="airplane"
        label={localeStrings.businessType}
        value={businessType?.replace('_', ' ')}
      />

      <AirlineRowItem icon="flag" label={localeStrings.isoCountryCode} value={isoCountry} />

      <AirlineRowItem icon="location" label={localeStrings.isoRegionCode} value={isoRegion} />

      <AirlineRowItem
        icon="star"
        label={localeStrings.skytraxRating}
        value={`${skytraxRating?.toFixed(1)} / 5`}
      />

      <AirlineRowItem icon="chatbox-ellipses" label={localeStrings.slogan} value={slogan} />
    </AirlineSectionRow>
  )
}
