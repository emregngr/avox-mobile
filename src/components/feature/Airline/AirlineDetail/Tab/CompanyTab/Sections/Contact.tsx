import { router } from 'expo-router'
import React, { useCallback, useMemo } from 'react'
import { Linking } from 'react-native'

import { AirportRowItem, AirportSectionRow } from '@/components/feature/Airport'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airline } from '@/types/feature/airline'

interface ContactProps {
  companyInfo: Airline['companyInfo']
  operations: Airline['operations']
}

export const Contact = ({ companyInfo, operations }: ContactProps) => {
  const { selectedLocale } = useLocaleStore()

  const { contactInfo, website } = companyInfo || {}
  const { region } = operations || {}
  const { email, phone } = contactInfo || {}

  const handlePhonePress = useCallback(async () => {
    if (phone) {
      await Linking?.openURL(`tel:${phone}`)
    }
  }, [phone])

  const handleEmailPress = useCallback(async () => {
    if (email) {
      await Linking?.openURL(`mailto:${email}`)
    }
  }, [email])

  const handleWebsitePress = useCallback(() => {
    if (website) {
      const formattedUrl = website.startsWith('http') ? website : `https://${website}`
      const regionLower = region?.toLowerCase()

      router.navigate({
        params: {
          regionLower,
          selectedWebsite: website,
          webViewUrl: formattedUrl,
        },
        pathname: '/web-view-modal',
      })
    }
  }, [website, region])

  const localeStrings = useMemo(
    () => ({
      contactInformation: getLocale('contactInformation'),
      email: getLocale('email'),
      phone: getLocale('phone'),
      website: getLocale('website'),
    }),
    [selectedLocale],
  )

  return (
    <AirportSectionRow title={localeStrings.contactInformation}>
      <AirportRowItem
        icon="globe"
        label={localeStrings.website}
        onPress={handleWebsitePress}
        value={website}
      />

      <AirportRowItem
        icon="call"
        label={localeStrings.phone}
        onPress={handlePhonePress}
        value={phone}
      />

      <AirportRowItem
        icon="mail"
        label={localeStrings.email}
        onPress={handleEmailPress}
        value={email}
      />
    </AirportSectionRow>
  )
}
