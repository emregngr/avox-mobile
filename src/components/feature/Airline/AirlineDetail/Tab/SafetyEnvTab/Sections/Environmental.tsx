import React from 'react'

import { ThemedText } from '@/components/common/ThemedText'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { SafetyHeader } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Cards/SafetyHeader'

interface EnvironmentalProps {
  content: string
  iconColor: string
  subtitle: string
  title: string
}

export const Environmental = ({ content, iconColor, subtitle, title }: EnvironmentalProps) => (
  <AirlineSectionRow title={title}>
    <SafetyHeader iconColor={iconColor} iconName="leaf" title={subtitle} />

    <ThemedText color="text-70" type="body2">
      {content}
    </ThemedText>
  </AirlineSectionRow>
)
