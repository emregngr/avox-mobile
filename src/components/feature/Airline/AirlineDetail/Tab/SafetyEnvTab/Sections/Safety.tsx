import React from 'react'

import { ThemedText } from '@/components/common/ThemedText'
import { AirlineSectionRow } from '@/components/feature/Airline/AirlineDetail/AirlineSectionRow'
import { CertificationsList } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Cards/CertificationsList'
import { SafetyHeader } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Cards/SafetyHeader'

interface SafetyProps {
  certifications: string[]
  certificationsTitle: string
  iconColor: string
  safetyRecord: string
  safetyRecordTitle: string
  title: string
}

export const Safety = ({
  certifications,
  certificationsTitle,
  iconColor,
  safetyRecord,
  safetyRecordTitle,
  title,
}: SafetyProps) => (
  <AirlineSectionRow title={title}>
    <SafetyHeader iconColor={iconColor} iconName="shield-check" title={safetyRecordTitle} />

    <ThemedText className="mb-4" color="text-70" type="body2">
      {safetyRecord}
    </ThemedText>

    <CertificationsList
      certifications={certifications}
      iconColor={iconColor}
      title={certificationsTitle}
    />
  </AirlineSectionRow>
)
