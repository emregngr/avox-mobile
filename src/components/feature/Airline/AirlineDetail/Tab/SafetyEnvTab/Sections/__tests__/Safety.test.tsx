import { render } from '@testing-library/react-native'
import React from 'react'

import { Safety } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Sections/Safety'

const mockedAirlineSectionRow = jest.fn()

jest.mock('@/components/common/ThemedText', () => {
  const { Text } = require('react-native')

  return {
    ThemedText: ({ children, ...props }: { children: string }) => (
      <Text {...props}>{children}</Text>
    ),
  }
})

jest.mock('@/components/feature/Airline/AirlineDetail/AirlineSectionRow', () => {
  const { View } = require('react-native')
  return {
    AirlineSectionRow: (props: any) => {
      mockedAirlineSectionRow(props)
      return (
        <View testID="mocked-section-row" {...props}>
          {props.children}
        </View>
      )
    },
  }
})

jest.mock('@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Cards/SafetyHeader', () => {
  const { View, Text } = require('react-native')
  return {
    SafetyHeader: ({
      title,
      iconName,
      iconColor,
    }: {
      title: string
      iconName: string
      iconColor: string
    }) => (
      <View data-icon-color={iconColor} data-icon-name={iconName} testID="mocked-safety-header">
        <Text>{title}</Text>
      </View>
    ),
  }
})

jest.mock(
  '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Cards/CertificationsList',
  () => {
    const { View, Text } = require('react-native')
    return {
      CertificationsList: ({
        title,
        certifications,
        iconColor,
      }: {
        title: string
        certifications: string[]
        iconColor: string
      }) => (
        <View
          data-cert-count={certifications.length}
          data-icon-color={iconColor}
          testID="mocked-certifications-list"
        >
          <Text>{title}</Text>
        </View>
      ),
    }
  },
)

const mockedDefaultProps = {
  title: 'Airline Safety Overview',
  safetyRecordTitle: 'Accident History',
  safetyRecord: 'This airline has maintained an impeccable safety record for over 15 years.',
  certificationsTitle: 'Global Certifications',
  certifications: ['IOSA Certified', 'EASA Approved'],
  iconColor: '#27ae60',
}

describe('Safety Component', () => {
  it('should pass the main title to the AirlineSectionRow component', () => {
    const { getByTestId } = render(<Safety {...mockedDefaultProps} />)

    const sectionRow = getByTestId('mocked-section-row')
    expect(sectionRow.props.title).toBe(mockedDefaultProps.title)
  })

  it('should render the safety record text', () => {
    const { getByText } = render(<Safety {...mockedDefaultProps} />)
    expect(getByText(mockedDefaultProps.safetyRecord)).toBeTruthy()
  })

  it('should pass the correct props to the SafetyHeader component', () => {
    const { getByTestId, getByText } = render(<Safety {...mockedDefaultProps} />)

    const header = getByTestId('mocked-safety-header')

    expect(getByText(mockedDefaultProps.safetyRecordTitle)).toBeTruthy()

    expect(header.props['data-icon-color']).toBe(mockedDefaultProps.iconColor)

    expect(header.props['data-icon-name']).toBe('shield-check')
  })

  it('should pass the correct props to the CertificationsList component', () => {
    const { getByTestId, getByText } = render(<Safety {...mockedDefaultProps} />)

    const list = getByTestId('mocked-certifications-list')

    expect(getByText(mockedDefaultProps.certificationsTitle)).toBeTruthy()

    expect(list.props['data-icon-color']).toBe(mockedDefaultProps.iconColor)
    expect(list.props['data-cert-count']).toBe(mockedDefaultProps.certifications.length)
  })
})

describe('Safety Component Snapshot', () => {
  it('should render the Safety Component successfully', () => {
    const { toJSON } = render(<Safety {...mockedDefaultProps} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
