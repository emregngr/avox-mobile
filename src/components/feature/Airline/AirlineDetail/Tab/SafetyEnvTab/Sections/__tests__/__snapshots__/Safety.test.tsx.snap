// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Safety Component Snapshot should render the Safety Component successfully 1`] = `
<View
  testID="mocked-section-row"
  title="Airline Safety Overview"
>
  <View
    data-icon-color="#27ae60"
    data-icon-name="shield-check"
    testID="mocked-safety-header"
  >
    <Text>
      Accident History
    </Text>
  </View>
  <Text
    className="mb-4"
    color="text-70"
    type="body2"
  >
    This airline has maintained an impeccable safety record for over 15 years.
  </Text>
  <View
    data-cert-count={2}
    data-icon-color="#27ae60"
    testID="mocked-certifications-list"
  >
    <Text>
      Global Certifications
    </Text>
  </View>
</View>
`;
